scripts/gradle/gradle-check.sh

```diff
@@ -50,7 +83,12 @@ if [ -z "$QUEUE_URL" ] || [ "$QUEUE_URL" != "null" ]; then
             echo "Jenkins Workflow Url: $WORKFLOW_URL"
             TIMEPASS=$(( TIMEPASS + 30 )) && echo time pass: $TIMEPASS
             sleep 30
-            RUNNING=$(curl -s -XGET ${WORKFLOW_URL}api/json | jq --raw-output .building)
+            RUNNING=$(perform_curl_and_process_with_jq "$WORKFLOW_URL" ".building" 5)
+            echo "Workflow running status :$RUNNING"
+            if [ $? -ne 0 ]; then
+                echo "Error: Failed to get or process response from Jenkins. Exiting."
+                exit 1
+            fi
```

Comment:
Suggest just exit the script when `if [ "$success" != true ]; then` during subroutine of `RUNNING=$(perform_curl_and_process_with_jq "$WORKFLOW_URL" ".building" 5)`, since `$?` is just checking the status of `echo`.

Reply:
right , let me correct that...thanks !!